### ISSUE TITLE
Support parsing H2's undocumented DROP TABLE .. CASCADE CONSTRAINTS syntax

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12829]
It appears that these two syntaxes do the same thing in H2:

```sql
DROP TABLE t CASCADE; -- Documented
DROP TABLE t CASCADE CONSTRAINTS; -- Undocumented
```

Probably some undocumented Oracle compatibility mode: https://docs.oracle.com/database/121/SQLRF/statements_9003.htm

**Comment by lukaseder** [comment]
Probably worth supporting in Oracle natively: https://github.com/jOOQ/jOOQ/issues/12830

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0 and 3.16.2 (https://github.com/jOOQ/jOOQ/issues/12831)
